test/test_functionalization.py

```diff
@@ -1542,6 +1543,87 @@ def forward(self, x_1):
     add = torch.ops.aten.add.Tensor(view_1, 1);  view_1 = None
     return add""")
 
+    def test_python_functionalization_zero_tensor(self):
+        def f(x):
+            y = torch.ops.aten._efficientzerotensor([4])
+            out = x + y
+            out.mul_(2)
+            return out
+        x = torch.randn(4)
+        out_ref = f(x)
+        out_test = dispatch_functionalize(f)(x)
+        out_test_cpp = _functionalize(f, reapply_views=True, crossref=False, skip_input_mutations=True)(x)
+        self.assertEqual(out_ref, out_test)
+        self.assertEqual(out_ref, out_test_cpp)
+        if not IS_WINDOWS:
+            fx_g = make_fx(dispatch_functionalize(f))(x)
+            fx_g_cpp = make_fx(_functionalize(f, reapply_views=True, crossref=False, skip_input_mutations=True))(x)
+            self.assertExpectedInline(fx_g.code.strip(), """\
+def forward(self, arg0_1):
+    _efficientzerotensor = torch.ops.aten._efficientzerotensor.default([4])
+    _to_copy = torch.ops.aten._to_copy.default(arg0_1, device = device(type='meta'), pin_memory = False)
+    zeros = torch.ops.aten.zeros.default([], dtype = torch.float32, layout = torch.strided, device = device(type='cpu'))
+    expand = torch.ops.aten.expand.default(zeros, [4]);  zeros = None
+    _to_copy_1 = torch.ops.aten._to_copy.default(expand, device = device(type='meta'), pin_memory = False);  expand = None
+    add = torch.ops.prims.add.default(_to_copy, _to_copy_1);  _to_copy = _to_copy_1 = None
+    expand_1 = torch.ops.aten.expand.default(arg0_1, [4]);  arg0_1 = None
+    _to_copy_2 = torch.ops.aten._to_copy.default(expand_1, dtype = torch.float32, layout = torch.strided, device = device(type='cpu'));  expand_1 = None
+    mul = torch.ops.aten.mul.Tensor(_to_copy_2, 2);  _to_copy_2 = None
+    return mul""")  # noqa: B950
+            self.assertEqual(fx_g_cpp.code.strip(), fx_g.code.strip())
+
+    def test_python_functionalization_conj(self):
+        def f(x):
+            y = x.conj()
+            y.mul_(2)
+            return torch.view_as_real(y.resolve_conj())
+
+        x = torch.randn(4, dtype=torch.complex64)
+        out_ref = f(x)
+        out_test = dispatch_functionalize(f)(x)
+        out_test_cpp = _functionalize(f, reapply_views=True, crossref=False, skip_input_mutations=True)(x)
+        # TODO: this is currently wrong, file an issue
+        # self.assertEqual(out_ref, out_test)
```

Comment:
Which one is wrong? What does it return?

Reply:
This is testing no functionalization vs. C++ functionalization.

So the C++ and python functionalization are both wrong (and give equally wrong results)